router/batchrouter/batchrouter.go

```diff
@@ -969,16 +975,23 @@ func (brt *HandleT) setJobStatus(batchJobs *BatchJobsT, isWarehouse bool, err er
 	jobRunIDAbortedEventsMap := make(map[string][]*router.FailedEventRowT)
 	var abortedEvents []*jobsdb.JobT
 	var batchReqMetric batchRequestMetric
-	if err != nil && err.Error() == DISABLED_EGRESS {
-		brt.logger.Debugf("BRT: Outgoing traffic disabled : %v at %v", batchJobs.BatchDestination.Source.ID, time.Now().Format("01-02-2006"))
-		batchJobState = jobsdb.Succeeded.State
-		errorResp = []byte(fmt.Sprintf(`{"success":"%s"}`, DISABLED_EGRESS))
-	} else if err != nil {
-		brt.logger.Errorf("BRT: Error uploading to object storage: %v %v", err, batchJobs.BatchDestination.Source.ID)
-		batchJobState = jobsdb.Failed.State
-		errorResp, _ = json.Marshal(ErrorResponseT{Error: err.Error()})
-		batchReqMetric.batchRequestFailed = 1
-		// We keep track of number of failed attempts in case of failure and number of events uploaded in case of success in stats
+	if err != nil {
+		switch err.Error() {
+		case DisabledEgress:
+			brt.logger.Debugf("BRT: Outgoing traffic disabled : %v at %v", batchJobs.BatchDestination.Source.ID, time.Now().Format("01-02-2006"))
+			batchJobState = jobsdb.Succeeded.State
+			errorResp = []byte(fmt.Sprintf(`{"success":"%s"}`, DisabledEgress))
+		case InvalidServiceProvider:
+			brt.logger.Debugf("BRT: Destination service provider not supported : %v at %v", batchJobs.BatchDestination.Source.ID, time.Now().Format("01-02-2006"))
```

Comment:
Log destination id and destination type as well.

Reply:
done